src/ucp/core/ucp_request.inl

```diff
@@ -226,31 +229,25 @@ ucp_request_can_complete_stream_recv(ucp_request_t *req)
  *         *req_status if filled with the completion status if completed.
  */
 static int UCS_F_ALWAYS_INLINE
-ucp_request_try_send(ucp_request_t *req, ucs_status_t *req_status,
-                     unsigned pending_flags)
+ucp_request_try_send(ucp_request_t *req, unsigned pending_flags)
 {
     ucs_status_t status;
 
     /* coverity wrongly resolves (*req).send.uct.func to test_uct_pending::pending_send_op_ok */
     /* coverity[address_free] */
     status = req->send.uct.func(&req->send.uct);
     if (status == UCS_OK) {
-        /* Completed the operation */
-        *req_status = UCS_OK;
+        /* Completed the operation, error also goes here */
```

Comment:
does it mean `req->send.uct.func` should always return OK in case of error?

Reply:
yes, to handle it from pending correctly, request contains actual status